backend/python/app/services/implementations/comment_service.py

```diff
@@ -0,0 +1,48 @@
+from datetime import datetime
+
+from flask import current_app
+from sqlalchemy import func
+
+from ...models import db
+from ...models.comment import Comment
+from ..interfaces.comment_service import ICommentService
+
+
+class CommentService(ICommentService):
+    def __init__(self, logger=current_app.logger):
+        self.logger = logger
+
+    def create_comment(self, comment):
+        try:
+            new_comment = Comment(**comment)
+            new_comment.time = datetime.utcnow()
+            new_comment.resolved = False
+            comment_index = 0
+
+            max = db.session.query(func.max(Comment.comment_index)).filter(
```

Comment:
out of curiosity: is a pattern like `Comment.query(...).filter(...)` possible in conjunction with using the `func.max`?

Reply:
I tried earlier but ran into an issue with aggregate functions ([here](https://stackoverflow.com/questions/41887460/select-list-is-not-in-group-by-clause-and-contains-nonaggregated-column-inc/41887524)), but wasn't sure if changing the sql mode variable was the best idea

Reply:
Interesting, good to know